src/lava/magma/core/run_configs.py

```diff
@@ -67,42 +67,62 @@ def select(self,
-                 select_sub_proc_model=False):
+    First, exceptions are checked in `exception_proc_model_map` dictionary
+    as key-value pairs `{Process: ProcessModel}`. These explicit
+    specifications are given precedence over the subsequent logic for
+    `ProcessModel` selection
+
+    If any `SubProcessModel`s are available, then they will be returned first.
+
+    In case of `PyProcessModel`s, an appropriate `PyProcessModel` is selected
+    based on the tags associated with it. Tags are set using the `@tag`
+    decorator. If no tags are set for `ProcessModel`s and no tag is selected,
+    then the first matching `ProcessModel` will be returned.
+    """
+
+    def __init__(self,
+                 custom_sync_domains=None,
+                 select_tag=None,
+                 select_sub_proc_model=False,
+                 exception_proc_model_map=None):
         super().__init__(custom_sync_domains=custom_sync_domains)
         self.select_tag = select_tag
         self.select_sub_proc_model = select_sub_proc_model
+        self.exception_proc_model_map = exception_proc_model_map
+        if not exception_proc_model_map:
+            self.exception_proc_model_map = {}
 
     def select(self, proc, proc_models):
         from lava.magma.core.model.sub.model import AbstractSubProcessModel
         from lava.magma.core.model.py.model import AbstractPyProcessModel
-        py_proc_model = None
-        sub_proc_model = None
+        # First priority to the exceptions. We will simply return the
+        # ProcessModel class associated with a Process class in the
+        # exceptions dictionary
+        if proc.__class__ in self.exception_proc_model_map:
+            return self.exception_proc_model_map[proc.__class__]
+        # Now, we loop over all ProcessModels proc_models of a Process proc
         for pm in proc_models:
-            if issubclass(pm, AbstractSubProcessModel):
-                sub_proc_model = pm
-            if issubclass(pm, AbstractPyProcessModel):
-                py_proc_model = pm
-                # Make selection
-            if self.select_sub_proc_model and sub_proc_model:
-                return sub_proc_model
-            elif py_proc_model:
-                if self.select_tag in py_proc_model.tags:
-                    return py_proc_model
-            else:
-                raise AssertionError("No legal ProcessModel found.")
+            # Priority given to SubProcessModels
+            if issubclass(pm, AbstractSubProcessModel) and \
+                    self.select_sub_proc_model:
+                return pm
+            elif issubclass(pm, AbstractPyProcessModel):
+                # If ProcessModel has tags AND user asked for a specific tag
+                if len(pm.tags) > 0 and (self.select_tag in pm.tags):
+                    return pm
+                # If ProcessModel has no tags and no one asked for a tag anyway
+                elif len(pm.tags) == 0 and not self.select_tag:
```

Comment:
If the user does not ask for a tag (which is the default in Loihi1SimCfg as select_tag=None), and the ProcessModel has at least one tag (like Dense), this currently gives the error:
```
AssertionError: No ProcessModel could be selected with tag 'None' for Process Process_17::Dense.
```
I would assume that some default ProcessModel is chosen if I don't care.

Reply:
I agree. If there is no ambiguity (not multiple ProcModels with same type for same Proc) then even if the ProcModel has tags but the user does not care about the tag, then we can just pick that one available option.

Ideally we could also detect when there is ambiguity (multiple ProcModels of same type for same Proc) and throw at least a warning when we pick the first ProcModel of many possible matching ones without a tag.

Reply:
Now we are checking if number of process models is 1. If it is, we just return that process model without any further checks. We print a warning when doing this, so that the user is aware.